cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go

```diff
@@ -1995,6 +1995,25 @@ func (p *BicepProvider) ensureParameters(
 			}
 		}
 
+		if param.IsAutoGen() {
+			// IsAutoGen already checked the azdMetadata exists
+			azdMetadata, _ := param.AzdMetadata()
+			genValue, err := autoGenerate(key, azdMetadata)
+			if err != nil {
+				return nil, err
+			}
+			configuredParameters[key] = azure.ArmParameterValue{
+				Value: genValue,
+			}
+			if err := p.env.Config.Set(configKey, genValue); err != nil {
```

Comment:
Does generate mean "generate once", or "always auto-generate"? I think setting it as config is the existing behavior, but just weird to see it here explicitly.

Reply:
I see now after expanding the few lines I was missing, that this does end making it remember. Can we add a comment here about `// we generate once, then save ii to config to be remembered next time` to codify? Makes it a lot easier to understand the intention.